src/components/PrintModal/WatermarkModal/WatermarkModal.js

```diff
@@ -77,28 +77,26 @@ class WatermarkModal extends React.PureComponent {
       isColorPaletteVisible: false,
       locationSettings,
       previousLocationSettings: locationSettings,
+      preExistingWatermark: {},
     };
     this.canvasContainerRef = React.createRef();
 
-    this.handleWatermarkRenderFxn = () => {
+    this.handleWatermarkRenderFxn = async () => {
       if (this.props.isVisible) {
+        const preExistingWatermark = await core.getWatermark();
         this.setState({
           locationSettings: this.state.previousLocationSettings,
+          preExistingWatermark,
         }, () => {
           this.addWatermarks();
         });
       } else {
-        this.removeAllWatermarks();
+        this.removeWatermarkCreatedByModal();
+        core.setWatermark(this.state.preExistingWatermarks);
       }
     };
   }
 
-  componentDidMount() {
-    if (this.props.isVisible !== undefined) {
-      core.addEventListener('documentLoaded', this.handleWatermarkRenderFxn);
```

Comment:
removed this.
from my distant memory, i wrote this to handle when watermark modal is opened on app loads.
it seems like:
- watermark modal can never be opened on app load
- it starting to look and feel weird that the watermark modal cares about the document being loaded

so i removed it for lesser complexity